### ISSUE TITLE
Super Table field data not showing in Preview (graphql) 

### ISSUE THREAD

**Issue by hannahwoodward** [engram-design/SuperTable#324]
Originally posted to Craft github, example of query & response can be found there - https://github.com/craftcms/cms/issues/5458

### Description
A client reported that preview wasn't updating for an entry (it was the first time they'd tried out preview, as we've only recently implemented it). On investigating, I found that everything except the super table field was updating correctly, and the super table field was returning an empty array. Querying for the entry normally (i.e. without preview query string) returns the super table field data as expected.

The site is headless, and the frontend is built using Nuxt & Graphql. The same problem is is happening both locally & on production. I've also tested the query on Postman (to see if it was a Nuxt/apollo issue), but I get the same response.

Originally noticed the issue on version 2.3.2, have since updated to v2.3.3.

### Steps to reproduce

1. Set up a headless site, using graphql to query data
2. Create Super Table field & add to a entry
3. When editing the entry, open in preview (or use share link)

### Additional info

- Craft version: 3.3.20.1
- PHP version: 7.3.4
- Database driver & version: 5.7.25
- Plugins & versions: Amazon S3 (1.2.7), Contact Form (2.2.6), Contact Form Honeypot (1.0.2), Expanded Singles (1.0.7), Redactor (2.4.0), SEOmatic (3.2.38), Super Table (2.3.2/2.3.3)

**Comment by hannahwoodward** [comment]
I've also tested on Super Table 2.3.0 & 2.3.1 and same issue occurs. Any update on this?

**Comment by bzin** [comment]
Having the same issue as @hannahwoodward with a setup with React and Next.js. 

All data returns correctly on a GraphQL preview query  apart of the Super Table field.

**Running:**
* Super Table 2.3.3 
* Craft 3.3.20.1

**Comment by engram-design** [comment]
@hannahwoodward @bzin I've not had experience with preview's yet, can you provide any further detail on your setup?

Have you run these queries in the GraphQL explorer? Do they produce expected results? My testing through this seems to be returning correct data, and have even used Insomnia/Postman to confirm.

**Comment by bzin** [comment]
@engram-design I have the same result using GraphQL Playground or using `fetch`. 

There is no error output, the data from the SuperTable field is just an empty array. I believe this is only occurring for `draft` entries that are not enabled/live.

But I found an exception. For example if I make sure that my revision dropdown is set to `Current` and I click on the `Share Preview` button I do see the page correctly. And the same does not occur if the revision dropdown as selected a draft.

**Comment by bzin** [comment]
Regarding the tech stack:

- Craft CMS is in `headless` mode
- Frontend is with Next.js and React
- Application is using SSR so I do request data often to Craft GraphQL API

**Comment by engram-design** [comment]
Ah, now thats interesting. There was issues in previous Craft 3 versions of this type of thing happening with Live Preview, but that was since resolved. I think it might be back for GQL queries. This only effected non-saved entries (drafts), hence it came up a lot in Live Preview, as you were editing field content, it wouldn't show up.

Does everything work okay for published entries?

**Comment by engram-design** [comment]
Think I've figured this one out. To get the fix early, change your `verbb/super-table` requirement in `composer.json` to:

```
"require": {
  "verbb/super-table": "dev-craft-3#479fe94d25d00bc0b7921b338973c3dd13390532 as 2.4.1.2",
  "...": "..."
}
```

Then run `composer update`.

**Comment by engram-design** [comment]
Fixed in [2.4.2](https://github.com/verbb/super-table/releases/tag/2.4.2)
